examples/block_writer/main.go

```diff
@@ -0,0 +1,212 @@
+	for i := range blockData {
+		blockData[i] = byte(bw.rand.Int() & 0xff)
+	}
+	return blockData
+}
+
+// parseFlags ensures that values taken from command line flags are valid.
+func parseFlags() error {
+	flag.Parse()
+	if *concurrency < 1 {
+		return util.Errorf("Value of 'concurrency' flag (%d) must be greater than or equal to 1", *concurrency)
+	}
+	if max, min := *maxBlockSizeBytes, *minBlockSizeBytes; max < min {
+		return util.Errorf("Value of 'max-block-bytes' (%d) must be greater than or equal to value of 'min-block-bytes' (%d)", max, min)
+	}
+	return nil
+}
+
+// maybeStartLocalServer will start a temporary local server if no database URL has been
+// provided.
+func maybeStartLocalServer() *server.TestServer {
+	if *dbURL != "" {
+		return nil
+	}
+	// Start a local test server.
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	*dbURL = "https://root@" + serv.ServingAddr() + "?certs=test_certs"
+	return serv
+}
+
+// setupDatabase performs initial setup for the example, creating a database and
+// with a single table. If the desired table already exists on the cluster, the
+// existing table will be dropped.
+func setupDatabase() (*sql.DB, error) {
+	parsedURL, err := url.Parse(*dbURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove database from parsedUrl
+	q := parsedURL.Query()
+	q.Del("database")
+	parsedURL.RawQuery = q.Encode()
+
+	// Open connection to server and create a database.
+	db, err := sql.Open("cockroach", parsedURL.String())
+	if err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+	if _, err := db.Exec("CREATE DATABASE IF NOT EXISTS datablocks"); err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+	db.Close()
+
+	// Open connection directly to the new database.
+	q.Add("database", "datablocks")
+	parsedURL.RawQuery = q.Encode()
+	db, err = sql.Open("cockroach", parsedURL.String())
+	if err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+	// Allow a maximum of concurrency+1 connections to the database.
+	db.SetMaxOpenConns(*concurrency + 1)
+
+	// Create the initial table for storing blocks.
+	if _, err := db.Exec(`DROP TABLE IF EXISTS blocks`); err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+	if _, err := db.Exec(`
+	CREATE TABLE IF NOT EXISTS blocks (
+	  writer_id STRING NOT NULL, 
+	  block_id BIGINT NOT NULL, 
+	  raw_bytes BYTES NOT NULL, 
+	  PRIMARY KEY (writer_id, block_id)
+	)`); err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+
+	return db, nil
+}
+
+func main() {
+	if err := parseFlags(); err != nil {
+		log.Fatal(err)
+	}
+
+	serv := maybeStartLocalServer()
+	if serv != nil {
+		defer serv.Stop()
+	}
+
+	db, err := setupDatabase()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	lastNow := time.Now()
+	var lastNumDumps uint64
+	writers := make([]blockWriter, *concurrency)
+
+	for i := range writers {
+		writers[i] = newBlockWriter(db)
+		go writers[i].run()
+	}
+
+	for range time.NewTicker(*outputInterval).C {
```

Comment:
Since you never stop the ticker this can be `for range time.Tick(*outputInterval)`